sys/kern/clock.c

```diff
@@ -4,21 +4,42 @@
 #include <sys/mimiker.h>
 #include <sys/klog.h>
 #include <sys/timer.h>
+#include <sys/context.h>
+#include <sys/gmon.h>
 
 static systime_t now = 0;
 static timer_t *clock = NULL;
-
 systime_t getsystime(void) {
   return now;
 }
 
+#if KPROF
+static void statclock(void) {
+  gmonparam_t *g = &_gmonparam;
+  thread_t *td = thread_self();
+  if (td->td_kframe == NULL)
+    return;
+  uintptr_t pc = ctx_get_pc(td->td_kframe), instr;
+  if (g->state == GMON_PROF_ON && pc >= g->lowpc) {
+    instr = pc - g->lowpc;
+    if (instr < g->textsize) {
+      instr /= HISTFRACTION * sizeof(*g->kcount);
+      g->kcount[instr]++;
+    }
+  }
+}
+#else
+static void statclock(void) {
+}
+#endif
```

Comment:
I think that you could use `__nothing` macro here like KASAN.

Reply:
Done.